test/extensions/transport_sockets/tls/tls_throughput_benchmark.cc

```diff
@@ -56,12 +54,11 @@ static void addFullSlices(Buffer::Instance& output_buffer, int num_slices, bool
 
   for (int i = 0; i < num_slices; i++) {
     auto start_size = buffer->length();
-    Buffer::RawSlice slices[2];
-    auto num_slices = buffer->reserve(16384, slices, 2);
-    for (unsigned i = 0; i < num_slices; i++) {
-      memset(slices[i].mem_, 'a', slices[i].len_);
+    Buffer::Reservation reservation = buffer->reserveApproximately(16384);
+    for (unsigned i = 0; i < reservation.numSlices(); i++) {
+      memset(reservation.slices()[i].mem_, 'a', reservation.slices()[i].len_);
     }
-    buffer->commit(slices, num_slices);
+    reservation.commit(16384);
```

Comment:
What guarantees that reserveApproximately returns at least 16kb?

Reply:
Changing it to use `reservation.size()`, but the next line will assert if it wasn't 16384. Given that it's just a test, I think this is good enough; it can be adjusted later if it starts failing.